great_expectations/rule_based_profiler/domain_builder/simple_semantic_type_domain_builder.py

```diff
@@ -122,52 +122,40 @@ def infer_semantic_domain_type_from_table_column_type(
-                    for type_name in ProfilerTypeMapping.FLOAT_TYPE_NAMES
-                ]
-            )
+            {type_name.upper() for type_name in ProfilerTypeMapping.INT_TYPE_NAMES}
+            | {type_name.upper() for type_name in ProfilerTypeMapping.FLOAT_TYPE_NAMES}
         ):
             semantic_column_type = SemanticDomainTypes.NUMERIC
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.STRING_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.STRING_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.TEXT
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.BOOLEAN_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.BOOLEAN_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.LOGIC
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.DATETIME_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.DATETIME_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.DATETIME
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.BINARY_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.BINARY_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.BINARY
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.CURRENCY_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.CURRENCY_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.CURRENCY
-        elif column_type in set(
-            [type_name.upper() for type_name in ProfilerTypeMapping.IDENTITY_TYPE_NAMES]
-        ):
+        elif column_type in {
+            type_name.upper() for type_name in ProfilerTypeMapping.IDENTITY_TYPE_NAMES
+        }:
             semantic_column_type = SemanticDomainTypes.IDENTITY
         elif column_type in (
-            set(
-                [
-                    type_name.upper()
-                    for type_name in ProfilerTypeMapping.MISCELLANEOUS_TYPE_NAMES
-                ]
-            )
-            | set(
-                [
-                    type_name.upper()
-                    for type_name in ProfilerTypeMapping.RECORD_TYPE_NAMES
-                ]
-            )
+            {
+                type_name.upper()
+                for type_name in ProfilerTypeMapping.MISCELLANEOUS_TYPE_NAMES
+            }
+            | {type_name.upper() for type_name in ProfilerTypeMapping.RECORD_TYPE_NAMES}
```

Comment:
Nice!

Reply:
You wrote this part so it sure is Nice! 😄